src/target/riscv/riscv-013.c

```diff
@@ -1759,6 +1821,9 @@ static int examine(struct target *target)
 		return ERROR_FAIL;
 	}
 
+	target->state = saved_tgt_state;
+	target->debug_reason = saved_dbg_reason;
+
 	if (!halted) {
 		riscv013_step_or_resume_current_hart(target, false);
 		target->state = TARGET_RUNNING;
```

Comment:
As far as I understand, this would work, but I'm not quite sure. My solution does not change the behavior, so it most definitely works. Please, see [this comment](https://github.com/riscv/riscv-openocd/pull/842#discussion_r1179414089).

Reply:
For clarity, let's continue the discussion about the target's state under the below comment.